### DISCUSSION TITLE
v1.6.x releases

### DISCUSSION THREAD

**Discussion by cx-miguel-dasilva** [Checkmarx/kics#5793] [Announcements]
# v1.6.0

# 🚀 New features and improvements
feat(knative&crossplane): add support to knative and crossplane (#5634)
feat(report): hide secrets in report results (#5504)
feat(scan): consider .gitignore to automatically exclude paths by default (#5506)
feat(pulumi): add support to Pulumi yaml parsing (#5648)
queries(pulumi): add pulumi gcp security queries (#5654)
queries(pulumi): add pulumi aws security queries (#5653)
queries(pulumi): add pulumi azure security queries (#5658)
feat(serverlessfw): add support to serverless fw  yml file parsing (#5670)
feat(knative): add knative security query and k8's pod queries interoperability (#5692)
feat(queires): add serverless framework queries (#5679)
feat(serverless): initial cloudformation security queries refactoring (#5697)
feat(engine): Kubernetes API support for runtime k8s clusters scan​ (#5651)

# 🐛 Bug fixes
fix(resolver): exclude resolve path call for the same path reference (#5511) (#5514)

# 📦 Dependency updates bumps
build(deps): bump github.com/zclconf/go-cty from 1.10.0 to 1.11.0
build(deps): bump github.com/aws/aws-sdk-go from 1.44.78 to 1.44.82
build(deps): bump github.com/moby/buildkit from 0.10.3 to 0.10.4
build(deps): bump helm.sh/helm/v3 from 3.9.3 to 3.9.4
ci(deps): bump goreleaser/goreleaser-action from 3.0.0 to 3.1.0
build(deps): bump github.com/mackerelio/go-osstat from 0.2.2 to 0.2.3
build(deps): bump github.com/hashicorp/hcl/v2 from 2.13.0 to 2.14.0
build(deps): bump github.com/tdewolff/minify/v2 from 2.12.0 to 2.12.1
build(deps): bump github.com/gookit/color from 1.5.1 to 1.5.2
build(deps): bump github.com/aws/aws-sdk-go from 1.44.82 to 1.44.90
build(deps): bump github.com/aws/aws-sdk-go from 1.44.90 to 1.44.91

# 👻 Maintenance
docs(kicsbot): update images digest

Contributors:

<hr /><em>This discussion was created from the release <a href='https://github.com/Checkmarx/kics/releases/tag/v1.6.0'>v1.6.0</a>.</em>

**Comment by gabriel-cx** [comment]
# v1.6.1

## 🚀 New features and improvements
added 2 queries for CloudFormation and Terraform

update(coverage): code coverage improvements (https://github.com/Checkmarx/kics/pull/5744)
feat(workflows): add workflow to check latest software versions (https://github.com/Checkmarx/kics/pull/5823)

## 🐛 Bug fixes
fix(query): fix query descriptionText for s3 logging disabled kms rotation and iam policies (https://github.com/Checkmarx/kics/pull/5810) by @tomk-orca
fix(query): fix queries expected value to 'should be...' (https://github.com/Checkmarx/kics/pull/5816) by @liorj-orca
fix(query): fix dockerfile security query regex (https://github.com/Checkmarx/kics/pull/5826)
fix(query): change s3 bucket acl grants write acp security query (https://github.com/Checkmarx/kics/pull/5780)
fix(query): remove string check in open api security query (https://github.com/Checkmarx/kics/pull/5831)
fix(query): change s3 bucket with all permissions security query (https://github.com/Checkmarx/kics/pull/5781)
fix(query): update s3 bucket policy accepts http requests security query (https://github.com/Checkmarx/kics/pull/5832)
fix(query): updated lambda_function_with_privileged_role (https://github.com/Checkmarx/kics/pull/5833)
fix(query): fix responses with wrong http status code security query (https://github.com/Checkmarx/kics/pull/5834)
fix(query): fixed Docker queries related to issues 5115, 5116, and 5118 (https://github.com/Checkmarx/kics/pull/5295)
fix(bug): bug in get metrics script (https://github.com/Checkmarx/kics/pull/5796)
fix(bug): add support for certificate body process from tfvar (https://github.com/Checkmarx/kics/pull/5837)
fix(terraform data source): added data resources resolver (https://github.com/Checkmarx/kics/pull/5839)

## 📦 Dependency updates bumps
build(deps): bump github.com/GoogleCloudPlatform/terraformer from 0.8.21 to 0.8.22 (https://github.com/Checkmarx/kics/pull/5817) by @tomk-orca
build(deps): bump github.com/spf13/viper from 1.12.0 to 1.13.0 (https://github.com/Checkmarx/kics/pull/5766)
build(deps): bump k8s.io/client-go from 0.24.3 to 0.25.1 (https://github.com/Checkmarx/kics/pull/5804)
build(deps): bump github.com/aws/aws-sdk-go from 1.44.91 to 1.44.101 (https://github.com/Checkmarx/kics/pull/5809)
build(deps): bump github.com/open-policy-agent/opa from 0.43.0 to 0.44.0 (https://github.com/Checkmarx/kics/pull/5777)

ci(deps): bump actions/upload-artifact from 2 to 3 (https://github.com/Checkmarx/kics/pull/5764)
ci(deps): bump golang from 1.19.0-alpine to 1.19.1-alpine (https://github.com/Checkmarx/kics/pull/5767)
ci(deps): bump docker/setup-buildx-action from 1 to 2 (https://github.com/Checkmarx/kics/pull/5770)

## 👻 Maintenance
chore(gitlab-ci): add --ci flag to gitlab examples (https://github.com/Checkmarx/kics/pull/5682) by @sluetze
update(docs): correct the GH action name (https://github.com/Checkmarx/kics/pull/5818) by @konstruktoid
update(docs): improve information in the configuration docs (https://github.com/Checkmarx/kics/pull/5829) by @VladMasarik
update(docs): update remediate docs (https://github.com/Checkmarx/kics/pull/5794)
update(docs): docker hub docs information update (https://github.com/Checkmarx/kics/pull/5800)
update(docs): community tab added into the docs.kics.io website (https://github.com/Checkmarx/kics/pull/5806)
update(docs): update information about github action versions (https://github.com/Checkmarx/kics/pull/5842)
update(workflows): gh action tag update for 1.6 kics version (https://github.com/Checkmarx/kics/pull/5841)
update(workflows): delete branching process for major versions (https://github.com/Checkmarx/kics/pull/5812)

**Comment by gabriel-cx** [comment]
# v1.6.2
## 🚀 New features and improvements
feat(bom): bill of materials for rds in aws cloudformation [#5856](https://github.com/Checkmarx/kics/pull/5856)
feat(bom): bill of material rds for terraform [#5843](https://github.com/Checkmarx/kics/pull/5843)
feat(bom): bill of materials for aws dynamodb [#5861](https://github.com/Checkmarx/kics/pull/5861)


## 🐛 Bug fixes
fix(query): correct GCP KMS crypto key rotation period queries + descriptions by [@Churro](https://github.com/Churro) in [#5863](https://github.com/Checkmarx/kics/pull/5863)
fix(query): terraform/aws/iam_access_key_is_exposed by [@jycamier](https://github.com/jycamier) in [#5846](https://github.com/Checkmarx/kics/pull/5846)
fix(query): fix false positive in aws_instance by [@patrickpichler](https://github.com/patrickpichler) in [#5903](https://github.com/Checkmarx/kics/pull/5903)
fix(query): remove redundant and flawed GCP KMS key rotation query by [@Churro](https://github.com/Churro) in [#5864](https://github.com/Checkmarx/kics/pull/5864)
fix(query): fix false positive for rds backup_retention_period not set by [@patrickpichler](https://github.com/patrickpichler) in [#5902](https://github.com/Checkmarx/kics/pull/5902)
fix community link for contribution [#5854](https://github.com/Checkmarx/kics/pull/5854)
fix(query): drop Configuration Aggregator to All Regions Disabled Security severity to MEDIUM by [@patrickpichler](https://github.com/patrickpichler) in [#5901](https://github.com/Checkmarx/kics/pull/5901)
fix(query): reduce NET_RAW capability not being dropped severity to MEDIUM by [@patrickpichler](https://github.com/patrickpichler) in [#5900](https://github.com/Checkmarx/kics/pull/5900)
fix(query): cover additional deprecated API versions in k8s rule by [@Churro](https://github.com/Churro) in [#5867](https://github.com/Checkmarx/kics/pull/5867)


## 📦 Dependency updates bumps
build(deps): bump github.com/tdewolff/minify/v2 from 2.12.1 to 2.12.2 [#5857](https://github.com/Checkmarx/kics/pull/5857)
build(deps): bump k8s.io/client-go from 0.25.1 to 0.25.2 [#5827](https://github.com/Checkmarx/kics/pull/5827)
build(deps): bump github.com/aws/aws-sdk-go from 1.44.101 to 1.44.107 [#5840](https://github.com/Checkmarx/kics/pull/5840)
build(deps): bump github.com/aws/aws-sdk-go from 1.44.107 to 1.44.109 [#5866](https://github.com/Checkmarx/kics/pull/5866)
build(deps): bump github.com/tdewolff/minify/v2 from 2.12.2 to 2.12.3 [#5868](https://github.com/Checkmarx/kics/pull/5868)
ci(deps): bump checkmarx/kics-action from 1.5 to 1.6 [#5852](https://github.com/Checkmarx/kics/pull/5852)
ci(deps): bump styfle/cancel-workflow-action from 0.10.0 to 0.10.1 [#5865](https://github.com/Checkmarx/kics/pull/5865)

## 👻 Maintenance
Add community meetings schedule & link [#5912](https://github.com/Checkmarx/kics/pull/5912)
docs(queries): update queries catalog [#5869](https://github.com/Checkmarx/kics/pull/5869)
docs(kicsbot): update images digest [#5853](https://github.com/Checkmarx/kics/pull/5853)



## New Contributors
* @patrickpichler made their first contribution in https://github.com/Checkmarx/kics/pull/5901

**Comment by gabriel-cx** [comment]
# v1.6.3
## 🚀 New features and improvements
* update(query): fixed typos in query folder name and query name in https://github.com/Checkmarx/kics/pull/5954


## 🐛 Bug fixes
* fix(query): Update Password And Secrets Security Query Documentation in https://github.com/Checkmarx/kics/pull/5938
* fix(ExpToString): fixed TraverseIndex evaluation in https://github.com/Checkmarx/kics/pull/5939
* fix(query): update CloudWatch Log Group Without KMS Security Query MetaData in https://github.com/Checkmarx/kics/pull/5943
* fix(query): readjusted "Memcached Disabled" to "Redis Disabled" in https://github.com/Checkmarx/kics/pull/5952
* fix(query): improved regex to find AWS Access Key in assets/queries/terraform/aws/hardcoded_aws_access_key_in_lambda in https://github.com/Checkmarx/kics/pull/5951
* fix(masked_secrets): Mask Secrets in All Vulnerability Preview in https://github.com/Checkmarx/kics/pull/5949

## 📦 Dependency updates bumps
* bump(deps): bump express, debug, and sentry-go in https://github.com/Checkmarx/kics/pull/5957
* bump(deps): express dependencies in https://github.com/Checkmarx/kics/pull/5962
* bump(deps): reverted debug and updated dependencies in https://github.com/Checkmarx/kics/pull/5963
* build(deps): bump github.com/tdewolff/minify/v2 from 2.12.3 to 2.12.4 in https://github.com/Checkmarx/kics/pull/5904
* docs(kicsbot): update images digest in https://github.com/Checkmarx/kics/pull/5906
* ci(deps): bump golang from 1.19.1-alpine to 1.19.2-alpine in https://github.com/Checkmarx/kics/pull/5909
* build(deps): bump github.com/aws/aws-sdk-go from 1.44.109 to 1.44.114 in https://github.com/Checkmarx/kics/pull/5914
* ci(deps): bump docker/build-push-action from 3.1.1 to 3.2.0 in https://github.com/Checkmarx/kics/pull/5924
* ci(deps): bump styfle/cancel-workflow-action from 0.10.1 to 0.11.0 in https://github.com/Checkmarx/kics/pull/5925
* ci(deps): bump docker/login-action from 2.0.0 to 2.1.0 in https://github.com/Checkmarx/kics/pull/5926
* build(deps): bump github.com/spf13/cobra from 1.5.0 to 1.6.0 in https://github.com/Checkmarx/kics/pull/5928
* build(deps): bump github.com/open-policy-agent/opa from 0.44.0 to 0.45.0 in https://github.com/Checkmarx/kics/pull/5929
* build(deps): bump k8s.io/apimachinery from 0.25.2 to 0.25.3 in https://github.com/Checkmarx/kics/pull/5933
* bump: updating software versions in https://github.com/Checkmarx/kics/pull/5918
* build(deps): bump github.com/aws/aws-sdk-go from 1.44.114 to 1.44.116 in https://github.com/Checkmarx/kics/pull/5936
* build(deps): bump golang.org/x/text from 0.3.7 to 0.3.8 in https://github.com/Checkmarx/kics/pull/5930
* build(deps): bump k8s.io/api from 0.25.2 to 0.25.3 in https://github.com/Checkmarx/kics/pull/5937
* build(deps): bump golang.org/x/text from 0.3.7 to 0.3.8 in https://github.com/Checkmarx/kics/pull/5940
* build(deps): bump k8s.io/client-go from 0.25.2 to 0.25.3 in https://github.com/Checkmarx/kics/pull/5941


## 👻 Maintenance
* docs(kicsbot): update images digest in https://github.com/Checkmarx/kics/pull/5931
* docs(kicsbot): update images digest in https://github.com/Checkmarx/kics/pull/5935
